

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>11. Differential expression analysis with DESeq2 &#8212; 2019-UNR-omics 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="12. Backup files" href="backup.html" />
    <link rel="prev" title="10. Read Quantification" href="rnaseq-quant.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="backup.html" title="12. Backup files"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="rnaseq-quant.html" title="10. Read Quantification"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2019-UNR-omics 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <ul>
<li><a class="reference internal" href="#">11. Differential expression analysis with DESeq2</a><ul>
<li><a class="reference internal" href="#move-salmon-output-quant-files-to-their-own-directory">11.1. Move salmon output quant files to their own directory</a></li>
<li><a class="reference internal" href="#copy-a-previously-made-gene-and-transcript-id-relationship-file-to-your-home-directory">11.2. Copy a previously-made gene and transcript id relationship file to your home directory</a></li>
<li><a class="reference internal" href="#grab-a-special-script-plotpcawithsamplenames-r">11.3. Grab a special script plotPCAWithSampleNames.R</a></li>
<li><a class="reference internal" href="#rstudio-reminder">11.4. Rstudio reminder</a></li>
<li><a class="reference internal" href="#working-in-rstudio">11.5. Working in Rstudio:</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="rnaseq-quant.html"
                        title="previous chapter">10. Read Quantification</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="backup.html"
                        title="next chapter">12. Backup files</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/DE.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="toc.html">Docs</a></li>
              
              <li>11. Differential expression analysis with DESeq2</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="differential-expression-analysis-with-deseq2">
<h1>11. Differential expression analysis with DESeq2<a class="headerlink" href="#differential-expression-analysis-with-deseq2" title="Permalink to this headline">¶</a></h1>
<p>Comparing gene expression differences in samples between experimental conditions.</p>
<p>We will be using <a class="reference external" href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>.</p>
<p>References:</p>
<ul class="simple">
<li><a class="reference external" href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">Documentation for DESeq2 with example analysis</a></li>
<li><a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">Love et al. 2014</a></li>
<li><a class="reference external" href="https://www.nature.com/nbt/journal/v34/n12/full/nbt.3682.html">Love et al. 2016</a></li>
</ul>
<p><a class="reference external" href="https://rnaseq-workshop-2017.readthedocs.io/en/latest/_static/Jane_differential_expression.pdf">DE lecture by Jane Khudyakov, July 2017</a></p>
<div class="section" id="move-salmon-output-quant-files-to-their-own-directory">
<h2>11.1. Move salmon output quant files to their own directory<a class="headerlink" href="#move-salmon-output-quant-files-to-their-own-directory" title="Permalink to this headline">¶</a></h2>
<p>RStudio only recognizes files in home <code class="docutils literal notranslate"><span class="pre">~/</span></code>. So, soft link files there:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">work</span>
<span class="n">mkdir</span> <span class="n">DE</span>
<span class="n">cd</span> <span class="n">DE</span>
<span class="n">mkdir</span> <span class="n">quant</span>
<span class="n">cd</span> <span class="n">quant</span>
<span class="n">ln</span> <span class="o">-</span><span class="n">s</span> <span class="o">~/</span><span class="n">work</span><span class="o">/</span><span class="n">rnaseq</span><span class="o">/*.</span><span class="n">quant</span> <span class="o">.</span>
</pre></div>
</div>
</div>
<div class="section" id="copy-a-previously-made-gene-and-transcript-id-relationship-file-to-your-home-directory">
<h2>11.2. Copy a previously-made gene and transcript id relationship file to your home directory<a class="headerlink" href="#copy-a-previously-made-gene-and-transcript-id-relationship-file-to-your-home-directory" title="Permalink to this headline">¶</a></h2>
<p>Generated with the <a class="reference external" href="https://angus.readthedocs.io/en/2018/dammit_annotation.html">dammit</a> annotation pipeline written by <a class="reference external" href="http://www.camillescott.org/dammit/">Camille Scott</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~</span>
<span class="n">cp</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">rnaseq</span><span class="o">/</span><span class="n">annotation</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span><span class="o">.</span><span class="n">dammit</span><span class="o">/</span><span class="n">nema_gene_name_id</span><span class="o">.</span><span class="n">csv</span> <span class="o">.</span>
</pre></div>
</div>
</div>
<div class="section" id="grab-a-special-script-plotpcawithsamplenames-r">
<h2>11.3. Grab a special script plotPCAWithSampleNames.R<a class="headerlink" href="#grab-a-special-script-plotpcawithsamplenames-r" title="Permalink to this headline">¶</a></h2>
<p>from <a class="reference external" href="https://med.nyu.edu/research/scientific-cores-shared-resources/applied-bioinformatics-laboratories/leadership">Igor Dolgalev</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~</span>
<span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">ngs</span><span class="o">-</span><span class="n">docs</span><span class="o">/</span><span class="mi">2017</span><span class="o">-</span><span class="n">dibsi</span><span class="o">-</span><span class="n">rnaseq</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="n">plotPCAWithSampleNames</span><span class="o">.</span><span class="n">R</span>
</pre></div>
</div>
</div>
<div class="section" id="rstudio-reminder">
<h2>11.4. Rstudio reminder<a class="headerlink" href="#rstudio-reminder" title="Permalink to this headline">¶</a></h2>
<p>Connect to RStudio (see <a class="reference external" href="https://setac-omics.readthedocs.io/en/latest/jetstream-bioconda-config.html">instructions for setting up password</a>)</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>echo http://$(hostname):8787/
</pre></div>
</div>
<p>Now go to that Web address in your Web browser, and log in with the username and password from above.</p>
</div>
<div class="section" id="working-in-rstudio">
<h2>11.5. Working in Rstudio:<a class="headerlink" href="#working-in-rstudio" title="Permalink to this headline">¶</a></h2>
<p>From this point on, we will be adding these commands into R Studio.</p>
<p>Load libraries</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="s2">&quot;lattice&quot;</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">tximport</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>
<span class="n">source</span><span class="p">(</span><span class="s1">&#39;~/plotPCAWithSampleNames.R&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Tell RStudio where your files are and ask whether they exist:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">setwd</span><span class="p">(</span><span class="s2">&quot;~/work/DE/quant/&quot;</span><span class="p">)</span>
<span class="nb">dir</span><span class="o">&lt;-</span><span class="s1">&#39;~/work/DE&#39;</span>
<span class="n">files_list</span> <span class="o">=</span> <span class="nb">list</span><span class="o">.</span><span class="n">files</span><span class="p">()</span>
<span class="n">files_list</span>
<span class="n">files</span> <span class="o">&lt;-</span> <span class="n">file</span><span class="o">.</span><span class="n">path</span><span class="p">(</span><span class="nb">dir</span><span class="p">,</span> <span class="s2">&quot;quant&quot;</span><span class="p">,</span><span class="n">files_list</span><span class="p">,</span> <span class="s2">&quot;quant.sf&quot;</span><span class="p">)</span>
<span class="n">names</span><span class="p">(</span><span class="n">files</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="s2">&quot;0Hour_1&quot;</span><span class="p">,</span><span class="s2">&quot;0Hour_2&quot;</span><span class="p">,</span><span class="s2">&quot;0Hour_3&quot;</span><span class="p">,</span><span class="s2">&quot;0Hour_4&quot;</span><span class="p">,</span><span class="s2">&quot;0Hour_5&quot;</span><span class="p">,</span><span class="s2">&quot;6Hour_1&quot;</span><span class="p">,</span><span class="s2">&quot;6Hour_2&quot;</span><span class="p">,</span><span class="s2">&quot;6Hour_3&quot;</span><span class="p">,</span><span class="s2">&quot;6Hour_4&quot;</span><span class="p">,</span><span class="s2">&quot;6Hour_5&quot;</span><span class="p">)</span>
<span class="n">files</span>
<span class="nb">print</span><span class="p">(</span><span class="n">file</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">files</span><span class="p">))</span>
</pre></div>
</div>
<p>Use the <code class="docutils literal notranslate"><span class="pre">nema_gene_name_id.csv</span></code> file to <a class="reference external" href="https://f1000research.com/articles/4-1521/v2">summarize expression at the gene level</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>tx2gene &lt;- read.csv(&quot;~/nema_gene_name_id.csv&quot;)
tx2gene &lt;- tx2gene[,c(4,3)]
cols&lt;-c(&quot;transcript_id&quot;,&quot;gene_id&quot;)
colnames(tx2gene)&lt;-cols
head(tx2gene)

txi.salmon &lt;- tximport(files, type = &quot;salmon&quot;, tx2gene = tx2gene,importer=read.delim)

head(txi.salmon$counts)
dim(txi.salmon$counts)
</pre></div>
</div>
<p>You should see output something like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="mi">136</span>  <span class="mi">10</span>
</pre></div>
</div>
<p>Assign experimental variables:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>condition = factor(c(&quot;0Hour&quot;,&quot;0Hour&quot;,&quot;0Hour&quot;,&quot;0Hour&quot;,&quot;0Hour&quot;,&quot;6Hour&quot;,&quot;6Hour&quot;,&quot;6Hour&quot;,&quot;6Hour&quot;,&quot;6Hour&quot;))
ExpDesign &lt;- data.frame(row.names=colnames(txi.salmon$counts), condition = condition)
ExpDesign
</pre></div>
</div>
<p>You should see output something like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>        <span class="n">condition</span>
<span class="mi">0</span><span class="n">Hour_1</span>     <span class="mi">0</span><span class="n">Hour</span>
<span class="mi">0</span><span class="n">Hour_2</span>     <span class="mi">0</span><span class="n">Hour</span>
<span class="mi">0</span><span class="n">Hour_3</span>     <span class="mi">0</span><span class="n">Hour</span>
<span class="mi">0</span><span class="n">Hour_4</span>     <span class="mi">0</span><span class="n">Hour</span>
<span class="mi">0</span><span class="n">Hour_5</span>     <span class="mi">0</span><span class="n">Hour</span>
<span class="mi">6</span><span class="n">Hour_1</span>     <span class="mi">6</span><span class="n">Hour</span>
<span class="mi">6</span><span class="n">Hour_2</span>     <span class="mi">6</span><span class="n">Hour</span>
<span class="mi">6</span><span class="n">Hour_3</span>     <span class="mi">6</span><span class="n">Hour</span>
<span class="mi">6</span><span class="n">Hour_4</span>     <span class="mi">6</span><span class="n">Hour</span>
<span class="mi">6</span><span class="n">Hour_5</span>     <span class="mi">6</span><span class="n">Hour</span>
</pre></div>
</div>
<p>Run DESeq2:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">dds</span> <span class="o">&lt;-</span> <span class="n">DESeqDataSetFromTximport</span><span class="p">(</span><span class="n">txi</span><span class="o">.</span><span class="n">salmon</span><span class="p">,</span> <span class="n">ExpDesign</span><span class="p">,</span> <span class="o">~</span><span class="n">condition</span><span class="p">)</span>
<span class="n">dds</span> <span class="o">&lt;-</span> <span class="n">DESeq</span><span class="p">(</span><span class="n">dds</span><span class="p">,</span> <span class="n">betaPrior</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<p>Get counts:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">counts_table</span> <span class="o">=</span> <span class="n">counts</span><span class="p">(</span> <span class="n">dds</span><span class="p">,</span> <span class="n">normalized</span><span class="o">=</span><span class="n">TRUE</span> <span class="p">)</span>
</pre></div>
</div>
<p>Filtering out low expression transcripts:</p>
<p>See plot from <a class="reference external" href="https://rnaseq-workshop-2017.readthedocs.io/en/latest/_static/Before-after_filter.pdf">Dr. Lisa Komoroske’s file</a> generated with <a class="reference external" href="https://www.bioconductor.org/help/workflows/RNAseq123/">RNAseq123</a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>filtered_norm_counts&lt;-counts_table[!rowSums(counts_table==0)&gt;=1, ]
filtered_norm_counts&lt;-as.data.frame(filtered_norm_counts)
GeneID&lt;-rownames(filtered_norm_counts)
filtered_norm_counts&lt;-cbind(filtered_norm_counts,GeneID)
dim(filtered_norm_counts)
head(filtered_norm_counts)
</pre></div>
</div>
<p>Estimate dispersion:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">plotDispEsts</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
</pre></div>
</div>
<p>This is not a regular dispersion plot. Why is that? Here is an <a class="reference external" href="https://github.com/ljcohen/ECE221_final_project/blob/master/DESeq2/Dispersion.png">example of a typical dispersion plot from DESeq2</a>.</p>
<p>PCA:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">log_dds</span><span class="o">&lt;-</span><span class="n">rlog</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
<span class="n">plotPCAWithSampleNames</span><span class="p">(</span><span class="n">log_dds</span><span class="p">,</span> <span class="n">intgroup</span><span class="o">=</span><span class="s2">&quot;condition&quot;</span><span class="p">,</span> <span class="n">ntop</span><span class="o">=</span><span class="mi">40000</span><span class="p">)</span>
</pre></div>
</div>
<p>Get DE results:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>res&lt;-results(dds,contrast=c(&quot;condition&quot;,&quot;6Hour&quot;,&quot;0Hour&quot;))
head(res)
res_ordered&lt;-res[order(res$padj),]
GeneID&lt;-rownames(res_ordered)
res_ordered&lt;-as.data.frame(res_ordered)
res_genes&lt;-cbind(res_ordered,GeneID)
dim(res_genes)
head(res_genes)
dim(res_genes)
res_genes_merged &lt;- merge(res_genes,filtered_norm_counts,by=unique(&quot;GeneID&quot;))
dim(res_genes_merged)
head(res_genes_merged)
res_ordered&lt;-res_genes_merged[order(res_genes_merged$padj),]
write.csv(res_ordered, file=&quot;nema_DESeq_all.csv&quot; )
</pre></div>
</div>
<p>Set a threshold cutoff of padj&lt;0.05:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>resSig = res_ordered[res_ordered$padj &lt; 0.05, ]
# threshold +- log2foldchange 1, but let&#39;s not do this - controversial
#resSig = resSig[resSig$log2FoldChange &gt; 1 | resSig$log2FoldChange &lt; -1,]
write.csv(resSig,file=&quot;nema_DESeq_padj0.05.csv&quot;)
</pre></div>
</div>
<p>MA plot with some gene names:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>plot(log2(res_ordered$baseMean), res_ordered$log2FoldChange, col=ifelse(res_ordered$padj &lt; 0.05, &quot;red&quot;,&quot;gray67&quot;),main=&quot;6 hr vs. 0 hr (padj&lt;0.05)&quot;,xlim=c(1,20),pch=20,cex=1,ylim=c(-12,12))
genes&lt;-resSig$GeneID
mygenes &lt;- resSig[,]
baseMean_mygenes &lt;- mygenes[,&quot;baseMean&quot;]
log2FoldChange_mygenes &lt;- mygenes[,&quot;log2FoldChange&quot;]
text(log2(baseMean_mygenes),log2FoldChange_mygenes,labels=genes,pos=2,cex=0.60)
</pre></div>
</div>
<p>Heatmap</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>d&lt;-resSig
dim(d)
head(d)
colnames(d)
d&lt;-d[,c(8:17)]
d&lt;-as.matrix(d)
d&lt;-as.data.frame(d)
d&lt;-as.matrix(d)
rownames(d) &lt;- resSig[,1]
head(d)

hr &lt;- hclust(as.dist(1-cor(t(d), method=&quot;pearson&quot;)), method=&quot;complete&quot;)
mycl &lt;- cutree(hr, h=max(hr$height/1.5))
clusterCols &lt;- rainbow(length(unique(mycl)))
myClusterSideBar &lt;- clusterCols[mycl]
myheatcol &lt;- colorRampPalette(c(&quot;red&quot;, &quot;green&quot;))(n = 75)
heatmap(d, main=&quot;6 hr vs. 0 hr (padj&lt;0.05)&quot;, 
          Rowv=as.dendrogram(hr),
          cexRow=0.75,cexCol=0.8,srtCol= 90,
          adjCol = c(NA,0),offsetCol=2.5, 
          Colv=NA, dendrogram=&quot;row&quot;, 
          scale=&quot;row&quot;, col=myheatcol, 
          density.info=&quot;none&quot;, 
          trace=&quot;none&quot;, RowSideColors= myClusterSideBar)
</pre></div>
</div>
<p>Additional links:</p>
<p><a class="reference external" href="http://htmlpreview.github.io/?https://github.com/ljcohen/Fhet_MDPL_MDPP_salinity_DE/blob/master/Fhet_MDPL_v_MDPP_interactiononly_FW_BW.html">Example DE analysis from two populations of killifish! (Fundulus heteroclitus MDPL vs. MDPL)</a></p>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="rnaseq-quant.html" title="previous chapter (use the left arrow)">10. Read Quantification</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="backup.html" title="next chapter (use the right arrow)">12. Backup files</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="backup.html" title="12. Backup files"
             >next</a> |</li>
        <li class="right" >
          <a href="rnaseq-quant.html" title="10. Read Quantification"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2019-UNR-omics 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2019, Sateesh Peri. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>