

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>6. Short read quality and trimming &#8212; 2018-setacna-rnaseq 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="7. Transcriptome assembly - some basics" href="transcriptome-assembly.html" />
    <link rel="prev" title="5. Jetstream: install bioconda on a blank Linux machine." href="jetstream-bioconda-config.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="transcriptome-assembly.html" title="7. Transcriptome assembly - some basics"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="jetstream-bioconda-config.html" title="5. Jetstream: install bioconda on a blank Linux machine."
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2018-setacna-rnaseq 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <ul>
<li><a class="reference internal" href="#">6. Short read quality and trimming</a><ul>
<li><a class="reference internal" href="#getting-started">6.1. Getting started</a></li>
<li><a class="reference internal" href="#data-source">6.2. Data source</a><ul>
<li><a class="reference internal" href="#linking-data-to-our-working-location">6.2.1. 1. Linking data to our working location</a></li>
<li><a class="reference internal" href="#fastqc">6.2.2. 2. FastQC</a></li>
<li><a class="reference internal" href="#trimmomatic">6.2.3. 3. Trimmomatic</a></li>
<li><a class="reference internal" href="#fastqc-again">6.2.4. 4. FastQC again</a></li>
<li><a class="reference internal" href="#multiqc">6.2.5. 5. MultiQc</a></li>
</ul>
</li>
</ul>
</li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="jetstream-bioconda-config.html"
                        title="previous chapter">5. Jetstream: install bioconda on a blank Linux machine.</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="transcriptome-assembly.html"
                        title="next chapter">7. Transcriptome assembly - some basics</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/quality-trimming.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="toc.html">Docs</a></li>
              
              <li>6. Short read quality and trimming</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="short-read-quality-and-trimming">
<span id="short-read-quality-and-trimming"></span><h1>6. Short read quality and trimming<a class="headerlink" href="#short-read-quality-and-trimming" title="Permalink to this headline">¶</a></h1>
<p>Learning objectives:</p>
<ul class="simple">
<li>Install software (fastqc, multiqc) via conda</li>
<li>download data</li>
<li>visualize read quality</li>
<li>quality filter and trim reads</li>
</ul>
<p>Start up a Jetstream m1.medium or larger
<a class="reference external" href="jetstream/boot.html">as per Jetstream startup instructions</a>.</p>
<hr class="docutils" />
<p>You should now be logged into your Jetstream computer!  You should see
something like this</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>titus@js-17-71:~$ 
</pre></div>
</div>
<div class="section" id="getting-started">
<span id="getting-started"></span><h2>6.1. Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h2>
<p>Change to your home directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span>
</pre></div>
</div>
<p>and install FastQC, MultiQC, and trimmomatic:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">y</span> <span class="n">fastqc</span> <span class="n">multiqc</span> <span class="n">trimmomatic</span>
</pre></div>
</div>
</div>
<div class="section" id="data-source">
<span id="data-source"></span><h2>6.2. Data source<a class="headerlink" href="#data-source" title="Permalink to this headline">¶</a></h2>
<p>Make a “data” directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span>
<span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="n">data</span>
<span class="n">cd</span> <span class="n">data</span>
</pre></div>
</div>
<p>and download download some data from the
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4878611/">Schurch et al, 2016</a>
yeast RNAseq study:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="mi">5</span><span class="n">daup</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="mi">8</span><span class="n">rvh5</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">ERR458494</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="mi">2</span><span class="n">wvn3</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">ERR458495</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="n">xju4a</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">ERR458500</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="n">nmqe6</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">ERR458501</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="n">qfsze</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">ERR458502</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>Let’s make sure we downloaded all of our data using <a class="reference external" href="https://en.wikipedia.org/wiki/Md5sum">md5sum</a>. An md5sum hash is a fingerprint that lets you compare with the original file. Some sequencing facilities will provide a file containing md5sum hashes for the files being delivered to you. By comparing your md5sum with the original md5sum generated by the sequencing facility, you can make sure you have downloaded your files completely. If there was a disruption that occurred during the download process, not all the bytes of the file will be transferred properly and you will likely be missing some of your sequences.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">md5sum</span> <span class="o">*.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>You should see this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">2</span><span class="n">b8c708cce1fd88e7ddecd51e5ae2154</span>  <span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="mi">36072</span><span class="n">a519edad4fdc0aeaa67e9afc73b</span>  <span class="n">ERR458494</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="mi">7</span><span class="n">a06e938a99d527f95bafee77c498549</span>  <span class="n">ERR458495</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="mi">107</span><span class="n">aad97e33ef1370cb03e2b4bab9a52</span>  <span class="n">ERR458500</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">fe39ff194822b023c488884dbf99a236</span>  <span class="n">ERR458501</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">db614de9ed03a035d3d82e5fe2c9c5dc</span>  <span class="n">ERR458502</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>To check whether your md5sum hashes match with a file containing md5sum hashes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">md5sum</span> <span class="o">*.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> <span class="o">&gt;</span> <span class="n">md5sum</span><span class="o">.</span><span class="n">txt</span>
<span class="n">md5sum</span> <span class="o">-</span><span class="n">c</span> <span class="n">err_md5sum</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>(First we’re creating a file containing md5sum of the files, then checking it. In reality, you wouldn’t be making this file to check. The facility would be creating it for you.)</p>
<p>You should see this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="n">OK</span>
<span class="n">ERR458494</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="n">OK</span>
<span class="n">ERR458495</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="n">OK</span>
<span class="n">ERR458500</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="n">OK</span>
<span class="n">ERR458501</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="n">OK</span>
<span class="n">ERR458502</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span><span class="p">:</span> <span class="n">OK</span>
</pre></div>
</div>
<p>Now if you type:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="n">l</span>
</pre></div>
</div>
<p>you should see something like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">titus</span> <span class="n">titus</span>  <span class="mi">59532325</span> <span class="n">Jun</span> <span class="mi">29</span> <span class="mi">09</span><span class="p">:</span><span class="mi">22</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">titus</span> <span class="n">titus</span>  <span class="mi">58566854</span> <span class="n">Jun</span> <span class="mi">29</span> <span class="mi">09</span><span class="p">:</span><span class="mi">22</span> <span class="n">ERR458494</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">titus</span> <span class="n">titus</span>  <span class="mi">58114810</span> <span class="n">Jun</span> <span class="mi">29</span> <span class="mi">09</span><span class="p">:</span><span class="mi">22</span> <span class="n">ERR458495</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">titus</span> <span class="n">titus</span> <span class="mi">102201086</span> <span class="n">Jun</span> <span class="mi">29</span> <span class="mi">09</span><span class="p">:</span><span class="mi">22</span> <span class="n">ERR458500</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">titus</span> <span class="n">titus</span> <span class="mi">101222099</span> <span class="n">Jun</span> <span class="mi">29</span> <span class="mi">09</span><span class="p">:</span><span class="mi">22</span> <span class="n">ERR458501</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">rw</span><span class="o">-</span><span class="n">r</span><span class="o">--</span> <span class="mi">1</span> <span class="n">titus</span> <span class="n">titus</span> <span class="mi">100585843</span> <span class="n">Jun</span> <span class="mi">29</span> <span class="mi">09</span><span class="p">:</span><span class="mi">22</span> <span class="n">ERR458502</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>These are six data files from the yeast study.
of the file.</p>
<p>One problem with these files is that they are writeable - by default,
UNIX makes things writeable by the file owner.  This poses an issue
with creating typos or errors in raw data.  Let’s fix that before we
go on any further:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">chmod</span> <span class="n">a</span><span class="o">-</span><span class="n">w</span> <span class="o">*</span>
</pre></div>
</div>
<p>Take a look at their permissions now –</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="n">l</span>
</pre></div>
</div>
<p>and you should see that the ‘w’ in the original permission string
(<code class="docutils literal notranslate"><span class="pre">-rw-rw-r--</span></code>) has been removed from each file and now it should look like <code class="docutils literal notranslate"><span class="pre">-r--r--r--</span></code>.</p>
<p>We’ll talk about what these files are below.</p>
<div class="section" id="linking-data-to-our-working-location">
<span id="linking-data-to-our-working-location"></span><h3>6.2.1. 1. Linking data to our working location<a class="headerlink" href="#linking-data-to-our-working-location" title="Permalink to this headline">¶</a></h3>
<p>First, make a new working directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="o">~/</span><span class="n">quality</span>
<span class="n">cd</span> <span class="o">~/</span><span class="n">quality</span>
</pre></div>
</div>
<p>Now, we’re going to make a “link” to our quality-trimmed data in our current working directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ln</span> <span class="o">-</span><span class="n">fs</span> <span class="o">~/</span><span class="n">data</span><span class="o">/*</span> <span class="o">.</span>
</pre></div>
</div>
<p>and you will see that they are now linked in the current directory when you do an
<code class="docutils literal notranslate"><span class="pre">ls</span></code>. These links save us from having to specify the full path (address) to their location on the computer, without us needing to actually move or copy the files. But note that changing these files here still changes the original files!</p>
<p>These are FASTQ files – let’s take a look at them:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">less</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>(use the spacebar to scroll down, and type ‘q’ to exit <code class="docutils literal notranslate"><span class="pre">less</span></code>)</p>
<p>Question:</p>
<ul class="simple">
<li>where does the filename come from?</li>
</ul>
<p>Links:</p>
<ul class="simple">
<li><a class="reference external" href="http://en.wikipedia.org/wiki/FASTQ_format">FASTQ Format</a></li>
</ul>
</div>
<div class="section" id="fastqc">
<span id="fastqc"></span><h3>6.2.2. 2. FastQC<a class="headerlink" href="#fastqc" title="Permalink to this headline">¶</a></h3>
<p>We’re going to use
<a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
summarize the data. We already installed ‘fastqc’ above, with the
conda command.</p>
<p>Now, run FastQC on two files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">fastqc</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">fastqc</span> <span class="n">ERR458500</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>Now type ‘ls’:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="n">d</span> <span class="o">*</span><span class="n">fastqc</span><span class="o">.</span><span class="n">zip</span><span class="o">*</span>
</pre></div>
</div>
<p>to list the files, and you should see:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ERR458493_fastqc</span><span class="o">.</span><span class="n">zip</span>
<span class="n">ERR458500_fastqc</span><span class="o">.</span><span class="n">zip</span>
</pre></div>
</div>
<p>Inside each of the fastqc directories you will find reports from the fastqc program. You can download these files using your RStudio Server console, if you like. (&#64;CTB)</p>
<p>or you can look at these copies of them:</p>
<ul class="simple">
<li><a class="reference external" href="http://htmlpreview.github.com/?https://github.com/ngs-docs/angus/blob/2018/_static/ERR458493_fastqc.html">ERR458493_fastqc.html</a></li>
<li><a class="reference external" href="http://htmlpreview.github.com/?https://github.com/ngs-docs/angus/blob/2018/_static/ERR458500_fastqc.html">ERR458500_fastqc.html</a></li>
</ul>
<p>Questions:</p>
<ul class="simple">
<li>What should you pay attention to in the FastQC report?</li>
</ul>
<p>Links:</p>
<ul class="simple">
<li><a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a class="reference external" href="http://www.youtube.com/watch?v=bz93ReOv87Y">FastQC tutorial video</a></li>
</ul>
<p>There are several caveats about FastQC - the main one is that it only
calculates certain statistics (like duplicated sequences) for subsets
of the data (e.g. duplicate sequences are only analyzed for the first
100,000 sequences in each file</p>
</div>
<div class="section" id="trimmomatic">
<span id="trimmomatic"></span><h3>6.2.3. 3. Trimmomatic<a class="headerlink" href="#trimmomatic" title="Permalink to this headline">¶</a></h3>
<p>Now we’re going to do some trimming!  We’ll be using
<a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>, which
(as with fastqc) we’ve already installed via conda.</p>
<p>The first thing we’ll need are the adapters to trim off:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cp</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda</span><span class="o">/</span><span class="n">pkgs</span><span class="o">/</span><span class="n">trimmomatic</span><span class="o">-*/</span><span class="n">share</span><span class="o">/</span><span class="n">trimmomatic</span><span class="o">-*/</span><span class="n">adapters</span><span class="o">/</span><span class="n">TruSeq2</span><span class="o">-</span><span class="n">PE</span><span class="o">.</span><span class="n">fa</span> <span class="o">.</span>
</pre></div>
</div>
<p>(you can look at the contents of this file with <code class="docutils literal notranslate"><span class="pre">cat</span> <span class="pre">TruSeq2-PE.fa</span></code>)</p>
<p>Now, to run Trimmomatic on both of them:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">trimmomatic</span> <span class="n">SE</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> \
        <span class="n">ERR458493</span><span class="o">.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> \
        <span class="n">ILLUMINACLIP</span><span class="p">:</span><span class="n">TruSeq2</span><span class="o">-</span><span class="n">PE</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="mi">2</span><span class="p">:</span><span class="mi">40</span><span class="p">:</span><span class="mi">15</span> \
        <span class="n">LEADING</span><span class="p">:</span><span class="mi">2</span> <span class="n">TRAILING</span><span class="p">:</span><span class="mi">2</span> \
        <span class="n">SLIDINGWINDOW</span><span class="p">:</span><span class="mi">4</span><span class="p">:</span><span class="mi">2</span> \
        <span class="n">MINLEN</span><span class="p">:</span><span class="mi">25</span>
        
<span class="n">trimmomatic</span> <span class="n">SE</span> <span class="n">ERR458500</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">gz</span> \
        <span class="n">ERR458500</span><span class="o">.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> \
        <span class="n">ILLUMINACLIP</span><span class="p">:</span><span class="n">TruSeq2</span><span class="o">-</span><span class="n">PE</span><span class="o">.</span><span class="n">fa</span><span class="p">:</span><span class="mi">2</span><span class="p">:</span><span class="mi">40</span><span class="p">:</span><span class="mi">15</span> \
        <span class="n">LEADING</span><span class="p">:</span><span class="mi">2</span> <span class="n">TRAILING</span><span class="p">:</span><span class="mi">2</span> \
        <span class="n">SLIDINGWINDOW</span><span class="p">:</span><span class="mi">4</span><span class="p">:</span><span class="mi">2</span> \
        <span class="n">MINLEN</span><span class="p">:</span><span class="mi">25</span>
        
</pre></div>
</div>
<p>You should see output that looks like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">...</span>
<span class="n">Input</span> <span class="n">Reads</span><span class="p">:</span> <span class="mi">1093957</span> <span class="n">Surviving</span><span class="p">:</span> <span class="mi">1092715</span> <span class="p">(</span><span class="mf">99.89</span><span class="o">%</span><span class="p">)</span> <span class="n">Dropped</span><span class="p">:</span> <span class="mi">1242</span> <span class="p">(</span><span class="mf">0.11</span><span class="o">%</span><span class="p">)</span>
<span class="n">TrimmomaticSE</span><span class="p">:</span> <span class="n">Completed</span> <span class="n">successfully</span>
</pre></div>
</div>
<p>We can also run the same process for all 6 samples more efficiently using a <code class="docutils literal notranslate"><span class="pre">for</span></code> loop, as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for filename in *.fastq.gz
do
        # first, make the base by removing fastq.gz
        base=$(basename $filename .fastq.gz)
        echo $base

        trimmomatic SE ${base}.fastq.gz \
                ${base}.qc.fq.gz \
                ILLUMINACLIP:TruSeq2-PE.fa:2:40:15 \
                LEADING:2 TRAILING:2 \
                SLIDINGWINDOW:4:2 \
                MINLEN:25
done
</pre></div>
</div>
<p>This script will go through each for the filenames that end with <code class="docutils literal notranslate"><span class="pre">fastq.gz</span></code> and run Trimmomatic for it.</p>
<p>Questions:</p>
<ul class="simple">
<li>How do you figure out what the parameters mean?</li>
<li>How do you figure out what parameters to use?</li>
<li>What adapters do you use?</li>
<li>What version of Trimmomatic are we using here? (And FastQC?)</li>
<li>Do you think parameters are different for RNAseq and genomic data sets?</li>
<li>What’s with these annoyingly long and complicated filenames?</li>
</ul>
<p>For a discussion of optimal trimming strategies, see
<a class="reference external" href="http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00013/abstract">MacManes, 2014</a> – it’s about RNAseq but similar arguments should apply to metagenome
assembly.</p>
<p>Links:</p>
<ul class="simple">
<li><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></li>
</ul>
</div>
<div class="section" id="fastqc-again">
<span id="fastqc-again"></span><h3>6.2.4. 4. FastQC again<a class="headerlink" href="#fastqc-again" title="Permalink to this headline">¶</a></h3>
<p>Run FastQC again on the trimmed files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">fastqc</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">fastqc</span> <span class="n">ERR458500</span><span class="o">.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<p>And now view my copies of these files:</p>
<ul class="simple">
<li><a class="reference external" href="http://htmlpreview.github.com/?https://github.com/ngs-docs/angus/blob/2018/_static/ERR458493.qc_fastqc.html">ERR458493.qc_fastqc.html</a></li>
<li><a class="reference external" href="http://htmlpreview.github.com/?https://github.com/ngs-docs/angus/blob/2018/_static/ERR458500.qc_fastqc.html">ERR458500.qc_fastqc.html</a></li>
</ul>
<p>Let’s take a look at the output files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">less</span> <span class="n">ERR458493</span><span class="o">.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>                                                       
</pre></div>
</div>
<p>(again, use spacebar to scroll, ‘q’ to exit less).</p>
</div>
<div class="section" id="multiqc">
<span id="multiqc"></span><h3>6.2.5. 5. MultiQc<a class="headerlink" href="#multiqc" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://multiqc.info/">MultiQC</a> aggregates results across many samples into a single report for easy comparison.</p>
<p>Run Mulitqc on both the untrimmed and trimmed files</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">multiqc</span> <span class="o">.</span>
</pre></div>
</div>
<p>And now you should see output that looks like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">This</span> <span class="ow">is</span> <span class="n">MultiQC</span> <span class="n">v1</span><span class="o">.</span><span class="mi">0</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Template</span>    <span class="p">:</span> <span class="n">default</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Searching</span> <span class="s1">&#39;.&#39;</span>
<span class="n">Searching</span> <span class="mi">15</span> <span class="n">files</span><span class="o">..</span>  <span class="p">[</span><span class="c1">####################################]  100%</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>          <span class="n">fastqc</span> <span class="p">:</span> <span class="n">Found</span> <span class="mi">4</span> <span class="n">reports</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Compressing</span> <span class="n">plot</span> <span class="n">data</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Report</span>      <span class="p">:</span> <span class="n">multiqc_report</span><span class="o">.</span><span class="n">html</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Data</span>        <span class="p">:</span> <span class="n">multiqc_data</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">MultiQC</span> <span class="n">complete</span>
</pre></div>
</div>
<p>You can view the output html file
<a class="reference external" href="http://htmlpreview.github.com/?https://github.com/ngs-docs/angus/blob/2018/_static/multiqc_report.html">multiqc_report.html</a> by going to RStudio, selecting the file, and saying “view in Web browser.”</p>
<p>Questions:</p>
<ul class="simple">
<li>is the quality trimmed data “better” than before?</li>
<li>Does it matter that you still have adapters!?</li>
</ul>
</div>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="jetstream-bioconda-config.html" title="previous chapter (use the left arrow)">5. Jetstream: install bioconda on a blank Linux machine.</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="transcriptome-assembly.html" title="next chapter (use the right arrow)">7. Transcriptome assembly - some basics</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="transcriptome-assembly.html" title="7. Transcriptome assembly - some basics"
             >next</a> |</li>
        <li class="right" >
          <a href="jetstream-bioconda-config.html" title="5. Jetstream: install bioconda on a blank Linux machine."
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2018-setacna-rnaseq 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2018, Elias Oziolor. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>