

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>6. Short read quality and trimming &#8212; 2019-UNR-omics 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="7. GitHub: How not to lose your entire analysis!" href="git_lesson.html" />
    <link rel="prev" title="5. Working with Bioconda." href="bioconda-config.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="git_lesson.html" title="7. GitHub: How not to lose your entire analysis!"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="bioconda-config.html" title="5. Working with Bioconda."
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2019-UNR-omics 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <ul>
<li><a class="reference internal" href="#">6. Short read quality and trimming</a><ul>
<li><a class="reference internal" href="#data-source">6.1. Data source</a></li>
<li><a class="reference internal" href="#check-that-your-data-is-where-it-should-be">6.2. Check that your data is where it should be</a></li>
<li><a class="reference internal" href="#quality-trimming-and-light-quality-filtering">6.3. Quality trimming and light quality filtering</a><ul>
<li><a class="reference internal" href="#link-your-data-into-your-working-directory">6.3.1. Link your data into your working directory</a></li>
<li><a class="reference internal" href="#fastqc">6.3.2. FastQC</a></li>
<li><a class="reference internal" href="#multiqc">6.3.3. MultiQC</a><ul>
<li><a class="reference internal" href="#view-your-files-on-your-own-computer">6.3.3.1. View your files on your own computer</a></li>
</ul>
</li>
<li><a class="reference internal" href="#adapter-trim-each-pair-of-files">6.3.4. Adapter trim each pair of files</a></li>
</ul>
</li>
</ul>
</li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="bioconda-config.html"
                        title="previous chapter">5. Working with Bioconda.</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="git_lesson.html"
                        title="next chapter">7. GitHub: How not to lose your entire analysis!</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/quality-trimming.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="toc.html">Docs</a></li>
              
              <li>6. Short read quality and trimming</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="short-read-quality-and-trimming">
<h1>6. Short read quality and trimming<a class="headerlink" href="#short-read-quality-and-trimming" title="Permalink to this headline">¶</a></h1>
<p>You should now be logged into your Atmosphere computer!  You should see
something like this</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>sateeshp@vm142-35:~$ 
</pre></div>
</div>
<p><strong>Note:</strong> If you haven’t added conda to your PATH yet, please do so by running the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>echo export PATH=$PATH:/opt/miniconda3/bin &gt;&gt; ~/.bashrc
</pre></div>
</div>
<p>Then, run the following command (or start a new terminal session) in order to activate the conda environment:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="o">~/.</span><span class="n">bashrc</span>
</pre></div>
</div>
<div class="section" id="data-source">
<h2>6.1. Data source<a class="headerlink" href="#data-source" title="Permalink to this headline">¶</a></h2>
<p>We will be using mRNAseq reads from a small subset of data from <a class="reference external" href="https://en.wikipedia.org/wiki/Starlet_sea_anemone">Nematostella vectensis</a> <a class="reference external" href="https://evodevojournal.biomedcentral.com/articles/10.1186/2041-9139-4-16">(Tulin et al., 2013)</a>.</p>
<p>Original RNAseq workflow protocol <a class="reference external" href="https://khmer-protocols.readthedocs.io/en/ctb/mrnaseq/">here</a>, more updated protocol <a class="reference external" href="http://eel-pond.readthedocs.io/en/latest/">here</a>.</p>
<p>Load your data into <code class="docutils literal notranslate"><span class="pre">work/data</span></code>.</p>
<p>First, create the subdirectories</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span>
<span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="n">work</span> <span class="n">work</span><span class="o">/</span><span class="n">data</span>
<span class="n">cd</span> <span class="n">work</span><span class="o">/</span><span class="n">data</span>
</pre></div>
</div>
<p>Download subset of data:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="mi">365</span><span class="n">fg</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">nema_subset_0Hour</span><span class="o">.</span><span class="n">zip</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">L</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">osf</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="mi">9</span><span class="n">tf2g</span><span class="o">/</span><span class="n">download</span> <span class="o">-</span><span class="n">o</span> <span class="n">nema_subset_6Hour</span><span class="o">.</span><span class="n">zip</span>
<span class="n">unzip</span> <span class="n">nema_subset_0Hour</span><span class="o">.</span><span class="n">zip</span>
<span class="n">unzip</span> <span class="n">nema_subset_6Hour</span><span class="o">.</span><span class="n">zip</span>
</pre></div>
</div>
<p>Define your $PROJECT variable to be the location of your work
directory; in this case, it will be <code class="docutils literal notranslate"><span class="pre">/mnt/work</span></code>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">export</span> <span class="n">PROJECT</span><span class="o">=~/</span><span class="n">work</span>
</pre></div>
</div>
<p>Now load your data in!</p>
</div>
<div class="section" id="check-that-your-data-is-where-it-should-be">
<h2>6.2. Check that your data is where it should be<a class="headerlink" href="#check-that-your-data-is-where-it-should-be" title="Permalink to this headline">¶</a></h2>
<p>Check:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ls $PROJECT/data
</pre></div>
</div>
<p>If you see all the files you think you should, good!  Otherwise, debug.</p>
<p>These are FASTQ files – let’s take a look at them:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">less</span> <span class="mi">0</span><span class="n">Hour_ATCACG_L002_R1_001</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
<p>(use the spacebar to scroll down, and type ‘q’ to exit ‘less’)</p>
<p>Question:</p>
<ul class="simple">
<li>where does the filename come from?</li>
<li>why are there 1 and 2 in the file names?</li>
</ul>
<p>Links:</p>
<ul class="simple">
<li><a class="reference external" href="http://en.wikipedia.org/wiki/FASTQ_format">FASTQ Format</a></li>
</ul>
</div>
<div class="section" id="quality-trimming-and-light-quality-filtering">
<h2>6.3. Quality trimming and light quality filtering<a class="headerlink" href="#quality-trimming-and-light-quality-filtering" title="Permalink to this headline">¶</a></h2>
<p>Make sure you’ve got the PROJECT location defined, and your data is there:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="nb">set</span> <span class="o">-</span><span class="n">u</span>
<span class="n">printf</span> <span class="s2">&quot;</span><span class="se">\n</span><span class="s2">My raw data is in $PROJECT/data/, and consists of $(ls -1 $</span><span class="si">{PROJECT}</span><span class="s2">/data/*.fastq | wc -l) files</span><span class="se">\n\n</span><span class="s2">&quot;</span>
<span class="nb">set</span> <span class="o">+</span><span class="n">u</span>
</pre></div>
</div>
<p><em>Important:</em> If you get an error above or the count of files is wrong…STOP!! Revisit the installation instructions!</p>
<div class="section" id="link-your-data-into-your-working-directory">
<h3>6.3.1. Link your data into your working directory<a class="headerlink" href="#link-your-data-into-your-working-directory" title="Permalink to this headline">¶</a></h3>
<p>Change into your project directory and make a workspace for quality trimming:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cd ${PROJECT}
mkdir -p quality
cd quality
</pre></div>
</div>
<p>Now, link the data files into your new workspace</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ln</span> <span class="o">-</span><span class="n">s</span> <span class="o">../</span><span class="n">data</span><span class="o">/*.</span><span class="n">fastq</span> <span class="o">.</span>
</pre></div>
</div>
<p>(Linking with <code class="docutils literal notranslate"><span class="pre">ln</span></code> avoids having to make a copy of the files, which will take up storage space.)</p>
<p>Check to make sure it worked</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">printf</span> <span class="s2">&quot;I see $(ls -1 *.fastq | wc -l) files here.</span><span class="se">\n</span><span class="s2">&quot;</span>
</pre></div>
</div>
<p>You can also do an <code class="docutils literal notranslate"><span class="pre">ls</span></code> to list the files.</p>
<p>If you see only one entry, <code class="docutils literal notranslate"><span class="pre">*.fastq</span></code>, then the ln command above didn’t work properly.  One possibility is that your files aren’t in your data directory; another is that their names don’t end with
<code class="docutils literal notranslate"><span class="pre">.fastq</span></code>.</p>
</div>
<div class="section" id="fastqc">
<h3>6.3.2. FastQC<a class="headerlink" href="#fastqc" title="Permalink to this headline">¶</a></h3>
<p>We’re going to use
<a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
summarize the data. We already installed ‘fastqc’ on our computer for
you.</p>
<p>Now, run FastQC on two files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">fastqc</span> <span class="o">*.</span><span class="n">fastq</span>
</pre></div>
</div>
<p>After this finishes running (has to run on each file so might take a while), type ‘ls’:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="n">d</span> <span class="o">*</span><span class="n">fastqc</span><span class="o">.</span><span class="n">zip</span><span class="o">*</span>
</pre></div>
</div>
<p>to list the files, and you should see a number of files with the extensions <code class="docutils literal notranslate"><span class="pre">.fastqc.zip</span></code>.</p>
<p>Inside each of the fatqc directories you will find reports from the fastqc. You can download these files using your RStudio Server console, if you like. To install and run an RStudio Server, go <a class="reference external" href="https://angus.readthedocs.io/en/2017/visualizing-blast-scores-with-RStudio.html#installing-and-running-rstudio-on-jetstream">here</a>.</p>
<p>Questions:</p>
<ul class="simple">
<li>What should you pay attention to in the FastQC report?</li>
<li>Which is “better”, file 1 or file 2? And why?</li>
</ul>
<p>Links:</p>
<ul class="simple">
<li><a class="reference external" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a class="reference external" href="http://www.youtube.com/watch?v=bz93ReOv87Y">FastQC tutorial video</a></li>
<li><a class="reference external" href="http://angus.readthedocs.io/en/2015/_static/2015-lecture2-sequencing.pptx.pdf">Examples of fastqc after technical sequencer problems</a>(starting on slide 40)</li>
</ul>
<p>There are several caveats about FastQC - the main one is that it only calculates certain statistics (like duplicated sequences) for subsets
of the data (e.g. duplicate sequences are only analyzed for the first 100,000 sequences in each file.</p>
</div>
<div class="section" id="multiqc">
<h3>6.3.3. MultiQC<a class="headerlink" href="#multiqc" title="Permalink to this headline">¶</a></h3>
<p>If you would like to aggregate all of your fastqc reports across many samples, <a class="reference external" href="http://multiqc.info/">MultiQC</a> will do this into a single report for easy comparison.</p>
<p>Run MultiQC:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">multiqc</span> <span class="o">.</span>
</pre></div>
</div>
<p>The terminal output should look like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">This</span> <span class="ow">is</span> <span class="n">MultiQC</span> <span class="n">v1</span><span class="o">.</span><span class="mi">6</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Template</span>    <span class="p">:</span> <span class="n">default</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Searching</span> <span class="s1">&#39;.&#39;</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>          <span class="n">fastqc</span> <span class="p">:</span> <span class="n">Found</span> <span class="mi">20</span> <span class="n">reports</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Compressing</span> <span class="n">plot</span> <span class="n">data</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Report</span>      <span class="p">:</span> <span class="n">multiqc_report</span><span class="o">.</span><span class="n">html</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">Data</span>        <span class="p">:</span> <span class="n">multiqc_data</span>
<span class="p">[</span><span class="n">INFO</span>   <span class="p">]</span>         <span class="n">multiqc</span> <span class="p">:</span> <span class="n">MultiQC</span> <span class="n">complete</span>
</pre></div>
</div>
<p>You can also view this output <a class="reference external" href="_static/multiqc_report.html">here</a></p>
<div class="section" id="view-your-files-on-your-own-computer">
<h4>6.3.3.1. View your files on your own computer<a class="headerlink" href="#view-your-files-on-your-own-computer" title="Permalink to this headline">¶</a></h4>
<p>As an alternative to viewing the files on the Rstudio server, we can secure copy (scp) these files to our own laptops, and view them from there.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="o">~/</span><span class="n">Desktop</span><span class="o">/</span><span class="n">nema_fastqc</span>  <span class="c1"># make a directory for these files</span>
<span class="n">scp</span> <span class="n">username</span><span class="nd">@ip</span><span class="o">.</span><span class="n">address</span><span class="p">:</span><span class="o">~/</span><span class="n">work</span><span class="o">/</span><span class="n">quality</span><span class="o">/*.</span><span class="n">html</span> <span class="o">~/</span><span class="n">Desktop</span><span class="o">/</span><span class="n">nema_fastqc</span>
</pre></div>
</div>
<p>where the first argument after <code class="docutils literal notranslate"><span class="pre">scp</span></code> is your login and path for the files we want to copy (from the atmosphere instance), and the second argument is the path to place the files on our own computer.</p>
<p>If you are unable to use scp though a terminal output, you can see the fastqc html output <a class="reference external" href="_static/6Hour_CGATGT_L002_R2_003.extract_fastqc.html">here</a></p>
</div>
</div>
<div class="section" id="adapter-trim-each-pair-of-files">
<h3>6.3.4. Adapter trim each pair of files<a class="headerlink" href="#adapter-trim-each-pair-of-files" title="Permalink to this headline">¶</a></h3>
<p>Setup trim directory:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">..</span>
<span class="n">mkdir</span> <span class="n">trim</span>
<span class="n">cd</span> <span class="n">trim</span>
<span class="n">ln</span> <span class="o">-</span><span class="n">s</span> <span class="o">../</span><span class="n">data</span><span class="o">/*.</span><span class="n">fastq</span> <span class="o">.</span>
<span class="n">cat</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda3</span><span class="o">/</span><span class="n">share</span><span class="o">/</span><span class="n">trimmomatic</span><span class="o">*/</span><span class="n">adapters</span><span class="o">/*</span> <span class="o">&gt;</span> <span class="n">combined</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>See excellent paper on trimming from <a class="reference external" href="http://journal.frontiersin.org/article/10.3389/fgene.2014.00013/full">MacManes 2014</a>.</p>
<p>Run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for filename in *_R1_*.fastq
do
# first, make the base by removing fastq
  base=$(basename $filename .fastq)
  echo $base
  
# now, construct the R2 filename by replacing R1 with R2
  baseR2=${base/_R1_/_R2_}
  echo $baseR2
        
# finally, run Trimmomatic
  trimmomatic PE ${base}.fastq ${baseR2}.fastq \
    ${base}.qc.fq.gz s1_se.gz \
    ${baseR2}.qc.fq.gz s2_se.gz \
    ILLUMINACLIP:combined.fa:2:40:15 \
    LEADING:2 TRAILING:2 \
    SLIDINGWINDOW:4:2 \
    MINLEN:25

# save the orphans
  zcat s1_se.gz s2_se.gz &gt;&gt; orphans.qc.fq.gz
  rm -f s1_se.gz s2_se.gz
done
</pre></div>
</div>
<p>Now, run fastqc again on trimmed files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">fastqc</span> <span class="o">*.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">multiqc</span> <span class="o">.</span>
</pre></div>
</div>
<p>The paired sequences output by this set of commands will be in the files ending in <code class="docutils literal notranslate"><span class="pre">.qc.fq.gz</span></code>, with any orphaned sequences all together
in <code class="docutils literal notranslate"><span class="pre">orphans.qc.fq.gz</span></code>.</p>
<p>Make these trimmed reads read-only and keep them, as we will reuse them later.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>chmod a-w ${PROJECT}/trim/*.qc.fq.gz
</pre></div>
</div>
<p>Questions:</p>
<ul class="simple">
<li>How do you figure out what the parameters mean?</li>
<li>How do you figure out what parameters to use?</li>
<li>What adapters do you use?</li>
<li>What version of Trimmomatic are we using here? (And FastQC?)</li>
<li>Do you think parameters are different for RNAseq and genomic data sets?</li>
<li>What’s with these annoyingly long and complicated filenames?</li>
<li>why are we running R1 and R2 together?</li>
</ul>
<p>For a discussion of optimal trimming strategies, see
<a class="reference external" href="http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00013/abstract">MacManes, 2014</a>
– it’s about RNAseq but similar arguments should apply to metagenome
assembly.</p>
<p>Links:</p>
<ul class="simple">
<li><a class="reference external" href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></li>
</ul>
<p>Questions:</p>
<ul class="simple">
<li>is the quality trimmed data “better” than before?</li>
<li>Does it matter that you still have adapters!?</li>
</ul>
</div>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="bioconda-config.html" title="previous chapter (use the left arrow)">5. Working with Bioconda.</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="git_lesson.html" title="next chapter (use the right arrow)">7. GitHub: How not to lose your entire analysis!</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="git_lesson.html" title="7. GitHub: How not to lose your entire analysis!"
             >next</a> |</li>
        <li class="right" >
          <a href="bioconda-config.html" title="5. Working with Bioconda."
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2019-UNR-omics 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2019, Elias Oziolor, Sateesh Peri. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>