

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  
  <!-- Licensed under the Apache 2.0 License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/open-sans/stylesheet.css" />
  <!-- Licensed under the SIL Open Font License -->
  <link rel="stylesheet" type="text/css" href="_static/fonts/source-serif-pro/source-serif-pro.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap.min.css" />
  <link rel="stylesheet" type="text/css" href="_static/css/bootstrap-theme.min.css" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
    <title>8. De novo transcriptome assembly &#8212; 2019-UNR-omics 0.0.1 documentation</title>
    <link rel="stylesheet" href="_static/guzzle.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="9. Read Quantification" href="rnaseq-quant.html" />
    <link rel="prev" title="7. GitHub: How not to lose your entire analysis!" href="git_lesson.html" />
  
   

  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="rnaseq-quant.html" title="9. Read Quantification"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="git_lesson.html" title="7. GitHub: How not to lose your entire analysis!"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2019-UNR-omics 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="container-wrapper">

      <div id="mobile-toggle">
        <a href="#"><span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span></a>
      </div>
  <div id="left-column">
    <div class="sphinxsidebar">
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <h2>Contents</h2>
    <div class="sidebar-localtoc">
      <ul>
<li><a class="reference internal" href="#">8. De novo transcriptome assembly</a><ul>
<li><a class="reference internal" href="#link-in-the-trimmed-data">8.1. Link in the trimmed data</a></li>
<li><a class="reference internal" href="#run-the-assembler">8.2. Run the assembler</a></li>
<li><a class="reference internal" href="#looking-at-the-assembly">8.3. Looking at the assembly</a></li>
<li><a class="reference internal" href="#suggestions-for-next-steps">8.4. Suggestions for next steps</a></li>
</ul>
</li>
</ul>

    </div>
  </div>
</div>
  <h4>Previous topic</h4>
  <p class="topless"><a href="git_lesson.html"
                        title="previous chapter">7. GitHub: How not to lose your entire analysis!</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="rnaseq-quant.html"
                        title="next chapter">9. Read Quantification</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/transcriptome-assembly.md.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div class="sidebar-block">
  <div class="sidebar-wrapper">
    <div id="main-search">
      <form class="form-inline" action="search.html" method="GET" role="form">
        <div class="input-group">
          <input name="q" type="text" class="form-control" placeholder="Search...">
        </div>
        <input type="hidden" name="check_keywords" value="yes" />
        <input type="hidden" name="area" value="default" />
      </form>
    </div>
  </div>
</div>
      
    </div>
  </div>
        <div id="right-column">
          
          <div role="navigation" aria-label="breadcrumbs navigation">
            <ol class="breadcrumb">
              <li><a href="toc.html">Docs</a></li>
              
              <li>8. De novo transcriptome assembly</li>
            </ol>
          </div>
          
          <div class="document clearer body">
            
  <div class="section" id="de-novo-transcriptome-assembly">
<h1>8. De novo transcriptome assembly<a class="headerlink" href="#de-novo-transcriptome-assembly" title="Permalink to this headline">¶</a></h1>
<p>Learning objectives:</p>
<ul class="simple">
<li>What is transcriptome assembly?</li>
<li>How do assemblers work?</li>
<li>Checking the quality of assembly</li>
<li>Understanding transcriptome assembly</li>
</ul>
<p>What if you are working with a species with no existing reference genome or transcriptome. How do you construct one?</p>
<p>The answer is <em>de novo</em> assembly. The basic idea with <em>de novo</em> transcriptome assembly is you feed in your reads and you get out a bunch of <em>contigs</em> that represent transcripts, or stretches of RNA present in the reads that don’t have any long repeats or much significant polymorphism. You run a  <em>de novo</em> transcriptome assembly program using the trimmed reads as input and get out a pile of assembled RNA.</p>
<p>Trinity, one of the leading <em>de novo</em> transcriptome assemblers, was developed at the <a class="reference external" href="http://www.broadinstitute.org/">Broad Institute</a> and the <a class="reference external" href="http://www.cs.huji.ac.il/">Hebrew University of Jerusalem</a>. We will be losely following steps from the <a class="reference external" href="https://eel-pond.readthedocs.io/en/latest">Eel pond protocol</a> for our guide to doing RNA-seq assembly.</p>
<div class="section" id="link-in-the-trimmed-data">
<h2>8.1. Link in the trimmed data<a class="headerlink" href="#link-in-the-trimmed-data" title="Permalink to this headline">¶</a></h2>
<p>We will be using the same set of <em>Nematostella vectensis</em> mRNAseq reads that we trimmed in the last lesson from <a class="reference external" href="https://evodevojournal.biomedcentral.com/articles/10.1186/2041-9139-4-16">Tulin et al., 2013</a>.</p>
<p>The following commands will create a new folder <code class="docutils literal notranslate"><span class="pre">assembly</span></code> and link the trimmed data we prepared earlier in the newly created folder:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cd ~/work
mkdir assembly
cd assembly

ln -fs ${PROJECT}/trim/*.qc.fq.gz .
ls
</pre></div>
</div>
<p>Combine all fq into 2 files (left.fq and right.fq)</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="o">*</span><span class="n">R1</span><span class="o">*.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="n">orphans</span><span class="o">.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">&gt;</span> <span class="n">left</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>
<span class="n">cat</span> <span class="o">*</span><span class="n">R2</span><span class="o">*.</span><span class="n">qc</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">&gt;</span> <span class="n">right</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</div>
<div class="section" id="run-the-assembler">
<h2>8.2. Run the assembler<a class="headerlink" href="#run-the-assembler" title="Permalink to this headline">¶</a></h2>
<p>Trinity works both with paired-end reads as well as single-end reads (including simultaneously both types at the same time). In the general case, the paired-end files are defined as <code class="docutils literal notranslate"><span class="pre">--left</span> <span class="pre">left.fq</span></code> and <code class="docutils literal notranslate"><span class="pre">--right</span> <span class="pre">right.fq</span></code> respectively. The single-end reads (a.k.a <em>orphans</em>) are defined by the flag <code class="docutils literal notranslate"><span class="pre">--single</span></code>.</p>
<p>So let’s run the assembler as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">time</span> <span class="n">Trinity</span> <span class="o">--</span><span class="n">seqType</span> <span class="n">fq</span> <span class="o">--</span><span class="n">max_memory</span> <span class="mi">15</span><span class="n">G</span> <span class="o">--</span><span class="n">CPU</span> <span class="mi">4</span> <span class="o">--</span><span class="n">left</span> <span class="n">left</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">right</span> <span class="n">right</span><span class="o">.</span><span class="n">fq</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">output</span> <span class="n">nema_trinity</span>
</pre></div>
</div>
<p>(This will take about 10 minutes)</p>
<p>You should see something like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">**</span> <span class="n">Harvesting</span> <span class="nb">all</span> <span class="n">assembled</span> <span class="n">transcripts</span> <span class="n">into</span> <span class="n">a</span> <span class="n">single</span> <span class="n">multi</span><span class="o">-</span><span class="n">fasta</span> <span class="n">file</span><span class="o">...</span>

<span class="n">Thursday</span><span class="p">,</span> <span class="n">October</span> <span class="mi">25</span><span class="p">,</span> <span class="mi">2018</span><span class="p">:</span> <span class="mi">21</span><span class="p">:</span><span class="mi">55</span><span class="p">:</span><span class="mi">15</span>	<span class="n">CMD</span><span class="p">:</span> <span class="n">find</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">read_partitions</span><span class="o">/</span> <span class="o">-</span><span class="n">name</span> <span class="s1">&#39;*inity.fasta&#39;</span>  <span class="o">|</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda3</span><span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">trinity</span><span class="o">-</span><span class="mf">2.8</span><span class="o">.</span><span class="mi">4</span><span class="o">/</span><span class="n">util</span><span class="o">/</span><span class="n">support_scripts</span><span class="o">/</span><span class="n">partitioned_trinity_aggregator</span><span class="o">.</span><span class="n">pl</span> <span class="o">--</span><span class="n">token_prefix</span> <span class="n">TRINITY_DN</span> <span class="o">--</span><span class="n">output_prefix</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">tmp</span>
<span class="o">-</span><span class="n">relocating</span> <span class="n">Trinity</span><span class="o">.</span><span class="n">tmp</span><span class="o">.</span><span class="n">fasta</span> <span class="n">to</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">Thursday</span><span class="p">,</span> <span class="n">October</span> <span class="mi">25</span><span class="p">,</span> <span class="mi">2018</span><span class="p">:</span> <span class="mi">21</span><span class="p">:</span><span class="mi">55</span><span class="p">:</span><span class="mi">15</span>	<span class="n">CMD</span><span class="p">:</span> <span class="n">mv</span> <span class="n">Trinity</span><span class="o">.</span><span class="n">tmp</span><span class="o">.</span><span class="n">fasta</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span>


<span class="c1">###################################################################</span>
<span class="n">Trinity</span> <span class="n">assemblies</span> <span class="n">are</span> <span class="n">written</span> <span class="n">to</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span>
<span class="c1">###################################################################</span>


<span class="n">Thursday</span><span class="p">,</span> <span class="n">October</span> <span class="mi">25</span><span class="p">,</span> <span class="mi">2018</span><span class="p">:</span> <span class="mi">21</span><span class="p">:</span><span class="mi">55</span><span class="p">:</span><span class="mi">15</span>	<span class="n">CMD</span><span class="p">:</span> <span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">miniconda3</span><span class="o">/</span><span class="n">opt</span><span class="o">/</span><span class="n">trinity</span><span class="o">-</span><span class="mf">2.8</span><span class="o">.</span><span class="mi">4</span><span class="o">/</span><span class="n">util</span><span class="o">/</span><span class="n">support_scripts</span><span class="o">/</span><span class="n">get_Trinity_gene_to_trans_map</span><span class="o">.</span><span class="n">pl</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span> <span class="o">&gt;</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">dibbears</span><span class="o">/</span><span class="n">work</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span><span class="o">.</span><span class="n">gene_trans_map</span>

<span class="n">real</span>	<span class="mi">7</span><span class="n">m7</span><span class="o">.</span><span class="mi">692</span><span class="n">s</span>
<span class="n">user</span>	<span class="mi">23</span><span class="n">m59</span><span class="o">.</span><span class="mi">929</span><span class="n">s</span>
<span class="n">sys</span>	<span class="mi">13</span><span class="n">m32</span><span class="o">.</span><span class="mi">485</span><span class="n">s</span>
</pre></div>
</div>
<p>at the end.</p>
</div>
<div class="section" id="looking-at-the-assembly">
<h2>8.3. Looking at the assembly<a class="headerlink" href="#looking-at-the-assembly" title="Permalink to this headline">¶</a></h2>
<p>First, save the assembly:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cp</span> <span class="n">nema_trinity</span><span class="o">/</span><span class="n">Trinity</span><span class="o">.</span><span class="n">fasta</span> <span class="n">nema</span><span class="o">-</span><span class="n">transcriptome</span><span class="o">-</span><span class="n">assembly</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>Now, look at the beginning:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">head</span> <span class="n">nema</span><span class="o">-</span><span class="n">transcriptome</span><span class="o">-</span><span class="n">assembly</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>These are the transcripts! Yay!</p>
<p>Let’s capture also some statistics of the Trinity assembly. Trinity provides a handy tool to do exactly that:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">TrinityStats</span><span class="o">.</span><span class="n">pl</span> <span class="n">nema</span><span class="o">-</span><span class="n">transcriptome</span><span class="o">-</span><span class="n">assembly</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>The output should look something like the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">################################</span>
<span class="c1">## Counts of transcripts, etc.</span>
<span class="c1">################################</span>
<span class="n">Total</span> <span class="n">trinity</span> <span class="s1">&#39;genes&#39;</span><span class="p">:</span>	<span class="mi">217</span>
<span class="n">Total</span> <span class="n">trinity</span> <span class="n">transcripts</span><span class="p">:</span>	<span class="mi">220</span>
<span class="n">Percent</span> <span class="n">GC</span><span class="p">:</span> <span class="mf">48.24</span>

<span class="c1">########################################</span>
<span class="n">Stats</span> <span class="n">based</span> <span class="n">on</span> <span class="n">ALL</span> <span class="n">transcript</span> <span class="n">contigs</span><span class="p">:</span>
<span class="c1">########################################</span>

	<span class="n">Contig</span> <span class="n">N10</span><span class="p">:</span> <span class="mi">1763</span>
	<span class="n">Contig</span> <span class="n">N20</span><span class="p">:</span> <span class="mi">819</span>
	<span class="n">Contig</span> <span class="n">N30</span><span class="p">:</span> <span class="mi">548</span>
	<span class="n">Contig</span> <span class="n">N40</span><span class="p">:</span> <span class="mi">407</span>
	<span class="n">Contig</span> <span class="n">N50</span><span class="p">:</span> <span class="mi">320</span>

	<span class="n">Median</span> <span class="n">contig</span> <span class="n">length</span><span class="p">:</span> <span class="mf">245.5</span>
	<span class="n">Average</span> <span class="n">contig</span><span class="p">:</span> <span class="mf">351.60</span>
	<span class="n">Total</span> <span class="n">assembled</span> <span class="n">bases</span><span class="p">:</span> <span class="mi">77353</span>


<span class="c1">#####################################################</span>
<span class="c1">## Stats based on ONLY LONGEST ISOFORM per &#39;GENE&#39;:</span>
<span class="c1">#####################################################</span>

	<span class="n">Contig</span> <span class="n">N10</span><span class="p">:</span> <span class="mi">1034</span>
	<span class="n">Contig</span> <span class="n">N20</span><span class="p">:</span> <span class="mi">605</span>
	<span class="n">Contig</span> <span class="n">N30</span><span class="p">:</span> <span class="mi">454</span>
	<span class="n">Contig</span> <span class="n">N40</span><span class="p">:</span> <span class="mi">357</span>
	<span class="n">Contig</span> <span class="n">N50</span><span class="p">:</span> <span class="mi">303</span>

	<span class="n">Median</span> <span class="n">contig</span> <span class="n">length</span><span class="p">:</span> <span class="mi">245</span>
	<span class="n">Average</span> <span class="n">contig</span><span class="p">:</span> <span class="mf">328.43</span>
	<span class="n">Total</span> <span class="n">assembled</span> <span class="n">bases</span><span class="p">:</span> <span class="mi">71270</span>
</pre></div>
</div>
<p>This is a set of summary stats about your assembly. Are they good? Bad? How would you know?</p>
</div>
<div class="section" id="suggestions-for-next-steps">
<h2>8.4. Suggestions for next steps<a class="headerlink" href="#suggestions-for-next-steps" title="Permalink to this headline">¶</a></h2>
<p>After generating a <em>de novo</em> transcriptome assembly:</p>
<ul class="simple">
<li><a class="reference external" href="https://angus.readthedocs.io/en/2018/dammit_annotation.html">annotation</a></li>
<li><a class="reference external" href="https://dibsi-rnaseq.readthedocs.io/en/latest/evaluation.html">evaluation</a></li>
</ul>
</div>
</div>


          </div>
            
  <div class="footer-relations">
    
      <div class="pull-left">
        <a class="btn btn-default" href="git_lesson.html" title="previous chapter (use the left arrow)">7. GitHub: How not to lose your entire analysis!</a>
      </div>
    
      <div class="pull-right">
        <a class="btn btn-default" href="rnaseq-quant.html" title="next chapter (use the right arrow)">9. Read Quantification</a>
      </div>
    </div>
    <div class="clearer"></div>
  
        </div>
        <div class="clearfix"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="rnaseq-quant.html" title="9. Read Quantification"
             >next</a> |</li>
        <li class="right" >
          <a href="git_lesson.html" title="7. GitHub: How not to lose your entire analysis!"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="toc.html">2019-UNR-omics 0.0.1 documentation</a> &#187;</li> 
      </ul>
    </div>
<script type="text/javascript">
  $("#mobile-toggle a").click(function () {
    $("#left-column").toggle();
  });
</script>
<script type="text/javascript" src="_static/js/bootstrap.js"></script>
  <div class="footer">
    &copy; Copyright 2019, Elias Oziolor, Sateesh Peri. Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
  </div>
  </body>
</html>